File: leetcode/python/4. Median of Two Sorted Arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        mergedList = []

        cycleLength = len(nums1) + len(nums2)
        nums1pos = 0
        nums2pos = 0
        
        for i in range(cycleLength):
            if nums1pos == len(nums1):
                for j in range(nums2pos, len(nums2)):
                    mergedList.append(nums2[j])
                break
            elif nums2pos == len(nums2):
                for j in range(nums1pos, len(nums1)):
                    mergedList.append(nums1[j])
                break

            if nums1[nums1pos] <= nums2[nums2pos]:
                mergedList.append(nums1[nums1pos])
                nums1pos += 1
            else:
                mergedList.append(nums2[nums2pos])
                nums2pos += 1
        
        if cycleLength % 2 == 0:
            return (mergedList[int(cycleLength / 2)] + mergedList[int(cycleLength / 2 - 1)]) / 2
        else:
            return mergedList[int((cycleLength - 1) / 2)]

File: leetcode/javascript/4. Median of Two Sorted Arrays.js
var findMedianSortedArrays = function(nums1, nums2) {
    const mergedList = [];

    const cycleLength = nums1.length + nums2.length;
    let nums1pos = 0;
    let nums2pos = 0;

    for (let i = 0; i < cycleLength; i += 1) {
        if (nums1pos === nums1.length) {
            for (let j = nums2pos; j < nums2.length; j += 1) {
                mergedList.push(nums2[j]);
            }
            break;
        } else if (nums2pos === nums2.length) {
            for (let j = nums1pos; j < nums1.length; j += 1) {
                mergedList.push(nums1[j]);
            }
            break;
        }

        if (nums1[nums1pos] <= nums2[nums2pos]) {
            mergedList.push(nums1[nums1pos]);
            nums1pos += 1;
        } else {
            mergedList.push(nums2[nums2pos]);
            nums2pos += 1;
        }
    }

    if (cycleLength % 2 === 0) {
        return (mergedList[cycleLength / 2] + mergedList[cycleLength / 2 - 1]) / 2;
    } else {
        return mergedList[(cycleLength - 1) / 2]
    }
};